src/sql/magic.py

```diff
@@ -406,22 +408,58 @@ def interactive_execute_wrapper(**kwargs):
         # only expand CTE if this is not a CTE itself
         if not is_cte:
             if args.with_:
+                query_type = (
+                    sqlparse.parse(command.sql)[0].get_type()
+                    if sqlparse.parse(command.sql)
+                    else None
+                )
```

Comment:
this same block (or similar one) is repeated in many places, can't we create a function? Also, I don't see any unit tests

Reply:
On it.